### PR TITLE
Adding react example for autocomplete and simple maps tutorial

### PR DESCRIPTION
**Pull request by vivekjain1397**
React example for Autocomplete component and a sample maps<>autocomplete implementation. 

https://github.com/user-attachments/assets/f2c1b2a3-5b3c-4a0c-bf0e-bb80898737d1

### CODE REVIEW

File: docs/maps/autocomplete.mdx [hunk 1]
```diff
@@ -101,3 +137,72 @@ export const App = () => (
...
+## Examples 
+
+
+### Use Autocomplete with Radar Maps in React
+
+Below is an example of how you can use [Radar Maps](https://radar.com/documentation/maps/maps) with Radar Autocomplete in React.
+```jsx
+import React, { useEffect, useRef } from 'react';
+import Radar from 'radar-sdk-js';
+import 'maplibre-gl/dist/maplibre-gl.css';
```
Comment (kochis) [inline docs/maps/autocomplete.mdx:149]: I don't think this is needed, it should already be getting pulled in from the Radar SDK. Unless you noticed that something specific was missing.

File: docs/maps/autocomplete.mdx [hunk 2]
```diff
@@ -81,6 +81,42 @@ Then, get started with the sample code below.
...
 
+### React 
+```jsx
+import React, { useEffect, useRef } from 'react';
+import Radar from 'radar-sdk-js';
+import 'radar-sdk-js/dist/radar.css';
+import AutocompleteUI from 'radar-sdk-js/dist/ui/autocomplete';
+
+const RadarAutocomplete = () => {
+const autocompleteRef = useRef<AutocompleteUI | null>(null);
```
Comment (kochis) [inline docs/maps/autocomplete.mdx:92]: indent

File: docs/maps/autocomplete.mdx [hunk 3]
```diff
@@ -101,3 +137,72 @@ export const App = () => (
...
+import 'radar-sdk-js/dist/radar.css';
+
+const RadarMap = () => {
+  const radarInitialized = useRef(false);
+  const mapRef = useRef(null);
+  const markerRef = useRef(null);
+
+  useEffect(() => {
+    if (radarInitialized.current) return;
+    radarInitialized.current = true;
```
Comment (kochis) [inline docs/maps/autocomplete.mdx:159]: What's this needed for?

File: docs/maps/autocomplete.mdx [hunk 4]
```diff
@@ -101,3 +137,72 @@ export const App = () => (
...
+
+        if (latitude && longitude) {
+          // Update marker position
+          markerRef.current.setLngLat([longitude, latitude]);
+
+          // Center the map on the selected address
+          mapRef.current.flyTo({ center: [longitude, latitude], zoom: 14 });
+        } else {
+          console.error('Coordinates are missing in the selected address:', address);
+        }
```
Comment (kochis) [inline docs/maps/autocomplete.mdx:194]: Is there ever a scenario where autocomplete returns without a lat/lng? If not, we should remove this.

File: docs/maps/autocomplete.mdx [hunk 5]
```diff
@@ -101,3 +137,72 @@ export const App = () => (
...
+      },
+    });
+  }, []);
+
+  return (
+    <div style={{ display: 'flex', height: '100vh' }}>
+      <div id="autocomplete" style={{ width: '400px', padding: '10px' }} />
+      <div id="map" style={{ flex: 1 }} />
+    </div>
```
Comment (kochis) [inline docs/maps/autocomplete.mdx:203]: Do you have a screenshot of what the map & autocomplete look like together?

File: docs/maps/autocomplete.mdx (hunk 3 above)
Comment (vivekjain1397) [reply docs/maps/autocomplete.mdx:159]: ah i was running into a random error so i added that check. Not needed now.

File: docs/maps/autocomplete.mdx (hunk 5 above)
Comment (vivekjain1397) [reply docs/maps/autocomplete.mdx:203]: Screenshot 2024-07-23 at 10 38 39 AM

It looks similar to our store locator demo we have right now.

File: docs/maps/autocomplete.mdx [hunk 6]
```diff
@@ -101,3 +136,68 @@ export const App = () => (
...
+
+        // Update marker position
+        markerRef.current.setLngLat([longitude, latitude]);
+
+        // Center the map on the selected address
+        mapRef.current.flyTo({ center: [longitude, latitude], zoom: 14 });
+      },
+    });
+    return () => {
+        autocompleteRef.current?.remove();
```
Comment (kochis) [inline docs/maps/autocomplete.mdx:190]: Lets also remove the map here too

File: docs/maps/autocomplete.mdx [hunk 7]
```diff
@@ -101,3 +136,68 @@ export const App = () => (
...
+import 'radar-sdk-js/dist/radar.css';
+
+const RadarMap = () => {
+  const radarInitialized = useRef(false);
+  const mapRef = useRef(null);
+  const markerRef = useRef(null);
+  const autocompleteRef = useRef(null);
+
+  useEffect(() => {
+    Radar.initialize('prj_test_pk_0f54bd903f71ab79eb24437b5e5c8c70346f911b');
```
Comment (kochis) [inline docs/maps/autocomplete.mdx:157]: Don't check this in

Comment (vivekjain1397) [reply docs/maps/autocomplete.mdx]: done.

File: docs/maps/autocomplete.mdx (hunk 6 above)
Comment (vivekjain1397) [reply docs/maps/autocomplete.mdx:190]: done.

**Review by kochis — APPROVED** [review]

**Pull request merged**